OpenRA.Mods.Common/Activities/DonateCash.cs

```diff
@@ -42,6 +42,12 @@ protected override void OnInside(Actor self)
 
 			if (self.Owner.IsAlliedWith(self.World.RenderPlayer))
 				self.World.AddFrameEndTask(w => w.Add(new FloatingText(target.CenterPosition, target.Owner.Color.RGB, FloatingText.FormatCashTick(payload), 30)));
+
+			foreach (var host in target.TraitsImplementing<INotifyCashTransfer>())
+				host.OnCashTransfer(target, self);
+
+			foreach (var me in self.TraitsImplementing<INotifyCashTransfer>())
+				me.OnCashTransfer(target, self);
```

Comment:
It is named like this one:
void OnCapture(Actor self, Actor captor, Player oldOwner, Player newOwner);